pygmt/tests/test_figure.py

```diff
@@ -292,6 +292,27 @@ def mock_psconvert(*args, **kwargs):  # pylint: disable=unused-argument
     }
 
 
+def test_figure_savefig_worldfile():
```

Comment:
Was there a reason to switch from the parametrized tests to a single test in a863132fb58629aa8efb1541ba88cbf7b13fcca4? It'll be harder to see which formats/extensions fail, if e.g. the first one in a list like ['.bmp', '.jpg', ...] raises an AssertionError, because the others in the list won't be tested.

Reply:
The main idea is to reducing the number of tests and also testing times. Comparing https://github.com/GenericMappingTools/pygmt/actions/runs/6677102075/job/18146781350?pr=2766 and https://github.com/GenericMappingTools/pygmt/actions/runs/6648648090/job/18065964154?pr=2766, the old way (using `parametrize`) has 7 tests for worldfile, and the testing time is 0.5 * 5 + 0.25 * 2 = 3 seconds, while the new way (a single test) costs 1.5 s only.

Reply:
Ah ok, I see that we avoid having to call `fig.basemap` multiple times, so it will be faster. All good then.